File: algorithms/array/two-pointers/code.cpp
#include <iostream>
#include <vector>
using namespace std;

pair<int, int> findPairWithSum(vector<int> &arr, int target) {
  int start = 0, end = arr.size() - 1;
  while (start < end) {
    int sum = arr[start] + arr[end];
    if (sum == target) {
      return {arr[start], arr[end]};
    } else if (sum < target) {
      start++;
    } else {
      end--;
    }
  }
  return {-1, -1}; // No pair found
}

int main() {
  vector<int> arr = {1, 2, 3, 4, 6, 8, 10};
  int target = 10;
  auto result = findPairWithSum(arr, target);
  if (result.first != -1) {
    cout << "Pair found: " << result.first << ", " << result.second << endl;
  } else {
    cout << "No pair found." << endl;
  }
  return 0;
}

File: algorithms/array/two-pointers/code.py
def find_pair_with_sum(arr, target):
    start, end = 0, len(arr) - 1
    while start < end:
        total = arr[start] + arr[end]
        if total == target:
            return (arr[start], arr[end])
        elif total < target:
            start += 1
        else:
            end -= 1
    return None  # No pair found

arr = [1, 2, 3, 4, 6, 8, 10]
target = 10
result = find_pair_with_sum(arr, target)
if result:
    print(f"Pair found: {result[0]}, {result[1]}")
else:
    print("No pair found.")
